.github/workflows/nightly-rust.yml

```diff
@@ -12,6 +12,8 @@ env:
 
 jobs:
   build-linux-with-rust-nightly:
+    # Run job only on servo/servo
+    if: github.repository == 'servo/servo'
```

Comment:
Should we have some easy way to deliberately allow these jobs on forks, e.g testing changes to the workflows? Maybe an additional check on presence of variable like 'RUN_ON_FORK'.

Reply:
While testing changes you can just temporary comment out those if guards, I believe that additional variables would make whole thing just more bloated and you would still need to change the value of variable, so changes are inevitable.

Reply:
The variable can be [defined at the repo level](https://docs.github.com/en/actions/learn-github-actions/variables#creating-configuration-variables-for-a-repository), without requiring changes to the guard. It can make things easier when creating PRs, but I have no strong opinion on this. I'll let other reviewers make a call.

Reply:
I completely forget that variables at repo level exist. But the problem is when you set it, it will apply for all workflows. This is not very hard problem, but it still can cause inconveniences.

Currently the workflows are blocked even if you manually dispatch it, so maybe the best compromise would be to allow workflows to be run when manually dispatched.

Reply:
> But the problem is when you set it, it will apply for all workflows

Will requiring the variable to specify the name of the workflow help? i.e The guard can be 
```
if: github.repository == 'servo/servo' || env.RUN_ON_FORK == github.workflow
```

Reply:
Hm, sure that could work, but I do not know if it is worth the bloat as workflow files are not changed often and someone may quickly forget about `RUN_ON_FORK` and what it does. Manual dispatch would be easier to work with it, more self-describing and what someone would expect to just-work™. The guard would be:
```
if: github.repository == 'servo/servo' || github.event_name == 'workflow_dispatch'
```